powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/controller/api/TemporaryKeyController.java

```diff
@@ -0,0 +1,83 @@
+ */
+
+package io.getlime.security.powerauth.app.server.controller.api;
+
+import com.wultra.security.powerauth.client.model.request.RemoveTemporaryPublicKeyRequest;
+import com.wultra.security.powerauth.client.model.request.TemporaryPublicKeyRequest;
+import com.wultra.security.powerauth.client.model.response.RemoveTemporaryPublicKeyResponse;
+import com.wultra.security.powerauth.client.model.response.TemporaryPublicKeyResponse;
+import io.getlime.core.rest.model.base.request.ObjectRequest;
+import io.getlime.core.rest.model.base.response.ObjectResponse;
+import io.getlime.security.powerauth.app.server.service.behavior.tasks.TemporaryKeyBehavior;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * Controller managing the endpoints related to temporary keys.
+ *
+ * @author Petr Dvorak, petr@wultra.com
+ */
+@RestController("temporaryKeyController")
+@RequestMapping("/rest/v3/keystore")
+@Tag(name = "PowerAuth ECIES Temporary Key Controller (V3)")
+@Slf4j
+public class TemporaryKeyController {
+
+    private final TemporaryKeyBehavior service;
+
+    @Autowired
+    public TemporaryKeyController(TemporaryKeyBehavior service) {
+        this.service = service;
+    }
+
+    /**
+     * Create temporary key.
+     *
+     * @param request Get temporary key parameters for given request.
+     * @return Response with temporary key.
+     * @throws Exception In case the service throws exception.
+     */
+    @PostMapping("/create")
+    public ObjectResponse<TemporaryPublicKeyResponse> createTemporaryKey(@RequestBody ObjectRequest<TemporaryPublicKeyRequest> request) throws Exception {
+        logger.info("TemporaryPublicKeyRequest received: {}", request);
+        final ObjectResponse<TemporaryPublicKeyResponse> response = new ObjectResponse<>(service.requestTemporaryKey(request.getRequestObject()));
+        logger.info("TemporaryPublicKeyRequest succeeded: {}", response);
+        return response;
+    }
+
+    /**
+     * Delete temporary key.
+     *
+     * @param request Delete temporary key with given ID.
+     * @return Response with deletion result.
+     */
+    @PostMapping("/remove")
+    public ObjectResponse<RemoveTemporaryPublicKeyResponse> deleteTemporaryKey(@RequestBody ObjectRequest<RemoveTemporaryPublicKeyRequest> request) {
+        logger.info("RemoveTemporaryPublicKeyRequest received: {}", request);
+        final ObjectResponse<RemoveTemporaryPublicKeyResponse> response = new ObjectResponse<>(service.removeTemporaryKey(request.getRequestObject()));
+        logger.info("RemoveTemporaryPublicKeyRequest succeeded: {}", response);
+        return response;
+    }
```

Comment:
This endpoint is not used in the restful integration project. Do we need it?

Reply:
This was intended to remove/expire the key by calling an API activly.